k2/csrc/ragged_ops.cu

```diff
@@ -1871,4 +1871,144 @@ Array1<int32_t> CoveringShapeForwardMap(RaggedShape &src,
   return ans;
 }
 
+
+
+void RaggedShapeAxis0Splitter::Init(RaggedShape &src) {
+  int32_t num_layers = src.NumLayers(),
+    num_layers_out = num_layers - 1,
+              dim0 = src.Dim0();
+  K2_CHECK(num_layers_out <= 4);  // If this fails, add something to the 4s and
```

Comment:
`K2_CHECK_LE` will give more detailed error info, i.e., it will print the value of `num_layers_out` on failure. `K2_CHECK` only prints `num_layers_out <= 4` but
does not tell us what the value `num_layer_out` is.